pkg/controllers/provisioningv2/rke2/machineprovision/controller.go

```diff
@@ -187,29 +195,44 @@ func (h *handler) getMachineStatus(job *batchv1.Job, create bool) (rkev1.RKEMach
 		}
 
 		if lastPod != nil {
-			return getMachineStatusFromPod(lastPod, create), nil
+			return getMachineStatusFromPod(lastPod, remove), nil
 		}
 	}
 
 	return rkev1.RKEMachineStatus{}, nil
 }
 
-func getMachineStatusFromPod(pod *corev1.Pod, create bool) rkev1.RKEMachineStatus {
+func getMachineStatusFromPod(pod *corev1.Pod, remove bool) rkev1.RKEMachineStatus {
+	reason := string(errors.CreateMachineError)
+	condType := createJobConditionType
+	if remove {
+		reason = string(errors.DeleteMachineError)
+		condType = deleteJobConditionType
+	}
+
 	if pod.Status.Phase == corev1.PodSucceeded {
 		return rkev1.RKEMachineStatus{
+			Conditions: []genericcondition.GenericCondition{
+				{
+					Type:   condType,
+					Status: corev1.ConditionTrue,
+				},
+			},
 			JobComplete: true,
 		}
 	}
 
 	for _, containerStatus := range pod.Status.ContainerStatuses {
 		if containerStatus.State.Terminated != nil && containerStatus.State.Terminated.ExitCode != 0 {
-			var reason string
-			if create {
-				reason = string(errors.CreateMachineError)
-			} else {
-				reason = string(errors.DeleteMachineError)
-			}
 			return rkev1.RKEMachineStatus{
+				Conditions: []genericcondition.GenericCondition{
+					{
+						Type:    condType,
+						Status:  corev1.ConditionFalse,
+						Reason:  reason,
+						Message: strings.TrimSpace(pod.Status.ContainerStatuses[0].State.Terminated.Message),
```

Comment:
Is it obvious that we only need message from the first container status when we're looping and returning only if exit code is non zero? Shouldn't this be `containerStatus.State.Terminated.Message`?

Reply:
@kinarashah Good point. I was just copying what was already there, but your suggestion makes more sense. I updated this.